src/modules/mavlink/mavlink_main.cpp

```diff
@@ -426,7 +426,7 @@ Mavlink::instance_exists(const char *device_name, Mavlink *self)
 	while (inst != nullptr) {
 
 		/* don't compare with itself */
-		if (inst != self && !strcmp(device_name, inst->_device_name)) {
+		if ((inst != self) && (inst->get_protocol() == SERIAL) && !strcmp(device_name, inst->_device_name)) {
```

Comment:
I think the fix is correct but I would suggest that the method `instance_exists` is renamed to `serial_instance_exists` and that the comment above is adapted.

Reply:
@julianoes 
I agree with you, it's a far better name.